src/Avalonia.Visuals/Animation/Keyframes/TransformKeyFrames.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Avalonia.Collections;
+using System.ComponentModel;
+using Avalonia.Animation.Utils;
+using System.Reactive.Linq;
+using System.Linq;
+using Avalonia.Media;
+
+namespace Avalonia.Animation.Keyframes
+{
+    /// <summary>
+    /// Key frames that handles <see cref="double"/> properties.
+    /// </summary>
+    public class TransformKeyFrames : KeyFrames<double>
+    {
+        DoubleKeyFrames childKeyFrames;
+
+        /// <inheritdoc/>
+        public override IDisposable Apply(Animation animation, Animatable control,  IObservable<bool> obsMatch)
+        {
+            var ctrl = (Visual)control;
+
+            // Check if the Target Property is Transform derived.
+            if (typeof(Transform).IsAssignableFrom(Property.OwnerType) && ctrl.RenderTransform != null)
+            {
+                var renderTransformType = ctrl.RenderTransform.GetType();
+
+                if (childKeyFrames == null)
+                {
+                    InitializeInternalDoubleKeyFrames();
+                }
+
+                // It's a transform object so let's target that.
+                if (renderTransformType == Property.OwnerType)
+                {
+                    return childKeyFrames.Apply(animation, ctrl.RenderTransform,  obsMatch);
+                }
+                // It's a TransformGroup and try finding the target there.
+                else if (renderTransformType == typeof(TransformGroup))
+                {
+                    foreach (Transform transform in ((TransformGroup)ctrl.RenderTransform).Children)
+                    {
+                        if (transform.GetType() == Property.OwnerType)
+                        {
+                            return childKeyFrames.Apply(animation, transform,  obsMatch);
+                        }
+                    }
+                }
+
+                //throw new InvalidOperationException($"TransformKeyFrame hasn't found an appropriate Transform object with type {Property.OwnerType} in target {control}.");
+
+                return null;
+
+            }
+            else
+            {
+                throw new Exception($"Unsupported property {Property}");
```

Comment:
Maybe be a bit more helpful here: `$"Cannot apply keyframe: property {Property} is not a Transform"` or something?